### ISSUE TITLE
Warnings are being hidden by the IDE for no reason

### ISSUE THREAD

**Issue by firewave** [johnthagen/clion-cppcheck#110]
```cpp
struct S1 {};

typedef struct S1 S2;

namespace N {

struct B {
  explicit B(int& i);
};

struct S2 : public B {
  explicit S2(int& i) : B(i) {}
};

} // namespace N
```

This should produce three warnings but only one (`missingReturn`) is being shown. They exist if you run the `Show Cppcheck XML Output` action.

If you make the code invalid and fix it you might be able to see the missing warnings for the blink of an eye if you are lucky.

I am using CLion 2026.1 Beta on Manjaro Linux.

**Comment by firewave** [comment]
The previous code no longer shows any warnings. Here's a different example:

```cpp
#include <cstdlib>

void f()
{
  int* p = (int*)malloc(0);
  p[1] = 0;
  free(p);
}
```
